src/rust-2018/edition-changes.md

```diff
@@ -0,0 +1,47 @@
+# 2018-Specific Changes
+
+The following is a summary of changes that only apply to code compiled with
+the 2018 edition compared to the 2015 edition.
+
+## Rust
+- [Non-lexical lifetimes].
```

Comment:
Might want to note that this is a temporary arrangement; we intend to backport later.